src/kernels/MIOpenLogCumSumExp.cpp

```diff
@@ -0,0 +1,386 @@
+                                                 const uint64_t reduce_size,
+                                                 const bool exclusive,
+                                                 const bool reverse)
+{
+    LogCumSumExpBackwardContiguousSmallCumDimStride1<FLOAT, REDUCE_SIZE>(
+        input, output, output_grad, input_grad, reduce_size, exclusive, reverse);
+}
+
+#else
+extern "C" __global__ void LogCumSumExpForwardSmallCumDim(const FLOAT* input,
+                                                          FLOAT* output,
+                                                          const uint64_t dim,
+                                                          const bool exclusive,
+                                                          const bool reverse,
+                                                          tensor_view_t<VIEW_DIMS> input_tv,
+                                                          tensor_view_t<VIEW_DIMS> output_tv)
+{
+    // instantiate the kernel
+    LogCumSumExpForwardSmallCumDim<FLOAT, VIEW_DIMS, REDUCE_SIZE>(
+        input, output, dim, exclusive, reverse, input_tv, output_tv);
+}
+
+extern "C" __global__ void LogCumSumExpBackwardSmallCumDim(const FLOAT* input,
+                                                           const FLOAT* output,
+                                                           const FLOAT* output_grad,
+                                                           FLOAT* input_grad,
+                                                           const uint64_t dim,
+                                                           const bool exclusive,
+                                                           const bool reverse,
+                                                           tensor_view_t<VIEW_DIMS> input_tv,
+                                                           tensor_view_t<VIEW_DIMS> output_tv,
+                                                           tensor_view_t<VIEW_DIMS> doutput_tv,
+                                                           tensor_view_t<VIEW_DIMS> dinput_tv)
+{
+    LogCumSumExpBackwardSmallCumDim<FLOAT, VIEW_DIMS, REDUCE_SIZE>(input,
+                                                                   output,
+                                                                   output_grad,
+                                                                   input_grad,
+                                                                   dim,
+                                                                   exclusive,
+                                                                   reverse,
+                                                                   input_tv,
+                                                                   output_tv,
+                                                                   doutput_tv,
+                                                                   dinput_tv);
+}
+
+#endif
```

Comment:
why you use header instead of specifying all functions at here? I see that those functions have different signatures.